src/index.js

```diff
@@ -37,11 +37,11 @@ root.render(
     {/* Use CSS baseline to set common spacing/sizing across all of our styling */}
     <CssBaseline />
     <BrowserRouter>
-      <AuthProvider {...authSettings}>
+      <RequireAuth {...authSettings}>
```

Comment:
This one should still be `AuthProvider`

`AuthProvider` is the "Context Provider" component that allows all the children of this element to have access to the useAuth hook. More about React Contexts here: https://react.dev/reference/react/createContext#provider

```suggestion
      <AuthProvider {...authSettings}>
```